src/app/application.cpp

```diff
@@ -771,6 +797,51 @@ void Application::shutdownCleanup(QSessionManager &manager)
 }
 #endif
 
+#ifdef Q_OS_WIN
+void Application::applyMemoryWorkingSetLimit()
+{
+//    MEMORYSTATUSEX statex;
+//    statex.dwLength = sizeof(statex);
+//    if (!::GlobalMemoryStatusEx(&statex))
+//    {
+//        const DWORD errorCode = ::GetLastError();
+//        QString reason = tr("Error %1.").arg(errorCode);
+//        LPVOID lpMsgBuf = nullptr;
+//        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+//                         , nullptr, ::GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+//        {
+//            reason = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
+//            ::LocalFree(lpMsgBuf);
+//        }
+//        LogMsg(tr("Failed to set Memory Working Set limit. Reason: %1").arg(reason), Log::WARNING);
+//        return;
+//    }
+
+    const int DEFAULT_WORKING_SET = 256;
+    const int MB = 1024 * 1024;
+    const std::int64_t limit = memoryWorkingSetLimit();
+    const SIZE_T maxSize = ((limit >= 0) ? limit : DEFAULT_WORKING_SET) * MB;
+    const SIZE_T minSize = (maxSize < (64 * MB)) ? maxSize : (64 * MB);
+    if (!::SetProcessWorkingSetSizeEx(::GetCurrentProcess(), minSize, maxSize, QUOTA_LIMITS_HARDWS_MAX_ENABLE))
+    {
+        const DWORD errorCode = ::GetLastError();
+        QString reason = tr("Error code: %1.").arg(errorCode);
+        LPVOID lpMsgBuf = nullptr;
+        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+                         , nullptr, errorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+        {
+            QString message = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
+            ::LocalFree(lpMsgBuf);
+
+            if (!message.endsWith(u'.'))
```

Comment:
And therefore this can be omitted.

Reply:
IMO, `Reason: Incorrect parameter. Error code: 87.` looks better than
`Error code: 87. Error message: "Incorrect parameter."` like you suggested. 
`:` is enough to provide good structure of messages in this case IMO. But you want to complicate it, although at the same time you object to the use of quotation marks (and other ways) to highlight some special words and phrases to emphasize that they are not used in the usual sense, but refer to certain terms, etc.

Reply:
>IMO, Reason: Incorrect parameter. Error code: 87. looks better than
Error code: 87. Error message: "Incorrect parameter." like you suggested.

What if the system (or other sources) error message have more than 1 sentence? the result log format will be messed up. Having it in quotes will clearly show the bounds of the `Reason: "msg".` part.

>you object to the use of quotation marks (and other ways) to highlight some special words and phrases to emphasize that they are not used in the usual sense

Right, those aren't "special" enough IMO. You don't see pompous highlighting on msdn doc or wikipedia entry or everywhere else.
* https://en.wikipedia.org/wiki/Capitalization_in_English#When_to_capitalize
  >with some nouns and adjectives, usually if a noun indicates a **proper noun**
* https://en.wikipedia.org/wiki/Quotation_marks_in_English
  >They also sometimes appear to be used as a means of adding emphasis, although this usage is usually considered incorrect.

Reply:
> What if the system (or other sources) error message have more than one sentences? the result log format will be messed up. Having it in quotes will clearly show the bounds of the `Reason: "msg".` part.

Therefore, I doubt about the unconditional unification of messages. In some cases (like this), the formation of less formalized (but more human like) messages looks better. For example:
>Failed to set physical memory (RAM) usage limit. Error code 87. Invalid parameter. Some additional error description.

Reply:
BTW, it is not some kind of "message from the depths of the system". It is just a string explanation of the error. We just use some predefined one provided by the WinAPI function instead of custom description.

Reply:
Other comments should be addressed now.

Reply:
I really don't want to argue about this part of the changes. If you insist, I will give in. But I would also like to hear other opinions (unless of course their waiting will not be long).

Reply:
>I really don't want to argue about this part of the changes. If you insist, I will give in.

I'll let you decide it yourself.

But now I would like to mention another issue of it.
Not every languages uses the (western?) punctuation marks. Some languages of Asia have their own (different) punctuation. For example: `。` instead of `.`. So here your detection won't be reliable for those languages.